datadog/resource_datadog_cloud_configuration_rule.go

```diff
@@ -0,0 +1,307 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/validators"
+)
+
+const nameField = "name"
+const messageField = "message"
+const enabledField = "enabled"
+const policyField = "policy"
+const resourceTypeField = "resource_type"
+const relatedResourceTypesField = "related_resource_types"
+const severityField = "severity"
+const notificationsField = "notifications"
+const groupByField = "group_by"
+const tagsField = "tags"
+
+func resourceDatadogCloudConfigurationRule() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Datadog Cloud Configuration Rule resource.",
+		CreateContext: cloudConfigurationRuleCreateContext,
+		ReadContext:   cloudConfigurationRuleReadContext,
+		UpdateContext: cloudConfigurationRuleUpdateContext,
+		DeleteContext: resourceDatadogSecurityMonitoringRuleDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: cloudConfigurationRuleSchema(),
+	}
+}
+
+func cloudConfigurationRuleSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
```

Comment:
Looks like missing a schema for `options`. The schema should resemble the spec. Also seem to be missing `cases`?

Reply:
No, this resource only focuses on `cloud_configuration` rules and we want to expose the fields only relevant to this type of rules. `cases` and `options` are handled internally when building or parsing the json payload.